### PR TITLE
[jbake] migrate authentication.md.vtl

### PR DESCRIPTION
**Pull request by fpapon**

### CODE REVIEW

**Review by bdemers — APPROVED** [review]
Added a couple of minor tweaks:
- added missing images
- removed code formatting from the header that contained `Subject`s

in: 56ec53bdaf08f89b66597ed98be881689d62eb3e

File: jbake/content/authentication.adoc [hunk 1]
```diff
@@ -0,0 +1,389 @@
...
+A subject is considered remembered if `subject.link:/static/current/apidocs/org/apache/shiro/subject/Subject.html#isRemembered()[isRemembered()]` returns `true`.
+
+* *Authenticated*: An authenticated `Subject` is one that has been successfully authenticated (i.e. the `login` method was invoked without throwing an exception) _during the Subject's current session_.
+A subject is considered authenticated if `subject.link:/static/current/apidocs/org/apache/shiro/subject/Subject.html#isAuthenticated()[isAuthenticated()]` returns `true`.
+
+[WARNING]
+====
+.Mutually Exclusive
+Remembered and authenticated states are mutually exclusive - a `true` value for one indicates a `false` value for the other and vice versa.
+====
```
Comment (bmarwell) [inline jbake/content/authentication.adoc:144]: According to https://asciidoctor.org/docs/asciidoc-writers-guide/#admonitions, I might argue that  `CAUTION` would be the better choice (now that we have it available):

* Use CAUTION to advise the reader to act carefully (i.e., exercise care).
* Use WARNING to inform the reader of danger, harm, or consequences that exist.

File: jbake/content/authentication.adoc [hunk 2]
```diff
@@ -0,0 +1,389 @@
...
+
+After a `Subject` logs-out, the `Subject` instance is considered anonymous again and, except for web applications, can be re-used for `login` again if desired.
+
+[CAUTION]
+====
+.Web Application Notice
+Because remembered identity in web applications is often persisted with cookies, and cookies can only be deleted before a Response body is committed, it is highly recommended to redirect the end-user to a new view or page immediately after calling `subject.logout()`.
+This guarantees that any security-related cookies are deleted as expected.
+This is a limitation of how HTTP cookies function and not a limitation of Shiro.
+====
```
Comment (bmarwell) [inline jbake/content/authentication.adoc:203]: Same, CAUTION might be a better choice which was not available in MD+SCMS.

File: jbake/content/authentication.adoc [hunk 3]
```diff
@@ -0,0 +1,389 @@
...
+
+*Step 4*: If more than one `Realm` is configured for the application, the `ModularRealmAuthenticator` instance will initiate a multi-`Realm` authentication attempt utilizing its configured link:/static/current/apidocs/org/apache/shiro/authc/pam/AuthenticationStrategy.html[`AuthenticationStrategy`].
+Before, during and after the `Realms` are invoked for authentication, the `AuthenticationStrategy` will be called to allow it to react to each Realm's results.
+We will cover `AuthenticationStrategies` soon.
+
+[CAUTION]
+====
+.Single-Realm Application
+If only a single Realm is configured, it is called directly - there is no need for an `AuthenticationStrategy` in a single-Realm application.
+====
```
Comment (bmarwell) [inline jbake/content/authentication.adoc:233]: This sounds more like INFO or TIP or NOTE to me.

**Review by bmarwell — CHANGES_REQUESTED** [review]
We should rethink the use of admonitions in general, while we are at it.

**Comment by fpapon** [comment]
> We should rethink the use of admonitions in general, while we are at it.

yes, for now I just convert from what we have currently in MD but we can re-think the level.

**Comment by fpapon** [comment]
> Added a couple of minor tweaks:
> 
>     * added missing images
> 
>     * removed code formatting from the header that contained `Subject`s
> 
> 
> in: [56ec53b](https://github.com/apache/shiro-site/commit/56ec53bdaf08f89b66597ed98be881689d62eb3e)

Thanks @bdemers I forget to check the image :(

**Comment by fpapon** [comment]
@bmarwell admonitions updated.

**Comment by bmarwell** [comment]
@fpapon thanks. I think it is good to take a quick look now, because it might be some time until we take a look at those paragraphs again!

**Review by bmarwell — APPROVED** [review]

**Pull request merged**